py/visdom/server.py

```diff
@@ -856,11 +856,10 @@ def wrap_func(handler, args):
         prev_eid = args.get('prev_eid')
         eid = args.get('eid')
 
-        if prev_eid not in handler.state:
-            return handler.write('TODO: env doesn\'t exit')
+        assert prev_eid in handler.state, 'env to be forked doesn\'t exit'
 
         handler.state[eid] = copy.deepcopy(handler.state[prev_eid])
```

Comment:
Just realizing you're missing a few small bits based on the implementation of fork that's present in SocketHandler:
```
eid = escape_eid(eid)
```
is used to make sure that the eid is valid before it's put in.
```
serialize_env(self.state, [eid], env_path=self.app.env_path)
```
ensures the new env is written to file once it's created, and 
```
broadcast_envs(self)
```
should ensure that open windows will receive the forked environment.

Reply:
Fixed :)